### DISCUSSION TITLE
How to switch focus to another window?

### DISCUSSION THREAD

**Discussion by govindn** [codeceptjs/CodeceptJS#3023] [General]
#### What are you trying to achieve?
I am trying to login to my app but via social login i.e. facebook or google

steps:
- open login url(where various login options present i.e. inbuilt login of app, login via facebook/google)
- click on 'login via facebook'
- a new window open where you have to enter facebook login id and password
- question is : how to switch focus to facebook login window now??
#### What do you get instead?
is there something like 'switchToNextTab' for switch window as well
> Provide console output if related. Use `--verbose` mode for more details.

```bash
# paste output here
```

> Provide test source code if related

```js
// paste test
```

### Details

* CodeceptJS version: 1.4.6
* NodeJS Version: 5.6.0
* Operating System: macOS
* Protractor || WebDriverIO || Nightmare version (if related) : protractor
* Configuration file:

```js
# paste suite config here
```

**Comment by kobenguyent** [comment]
@govindn have you tried https://codecept.io/helpers/WebDriver#switchtonexttab

**Comment by GSasu** [comment]
@govindn In case the solution proposed by @PeterNgTr is not sufficient I would suggest having a look over wdio switchWindow and implement it through a helper
https://webdriver.io/docs/api/browser/switchWindow.html

**Comment by jim-crawford** [comment]
After 2 and a bit years I would hope this has been addressed. Has there been any updates on this? I too have a similar issue but in  this case it is making a purchase from a third party checkout window, the basket page greyed out and a new window opens with the checkout form in it, how can you move focus from the background window to the foreground window?

**Reply by GSasu** [reply]
Have you tried [this](https://codecept.io/helpers/WebDriver/#switchtowindow)? Or the suggestion [here](https://github.com/codeceptjs/CodeceptJS/discussions/3023#discussioncomment-1283129)?
